app/views/users/show.html.erb

```diff
@@ -1 +1,23 @@
-<h1>Details of a selected user</h1>
\ No newline at end of file
+<section class="container">
+<article class="users-article">
+<div class="photo"><%= @user.photo %></div>
+<div class="details">
+ <h2><%= @user.name %></h2>
+  <p><%= link_to "Number of posts:", user_posts_url(@user.id) %> <%= @user.posts_counter %></p>
+</div>
+</article>
+<div class="bio">
+  <p>Bio</p>
+  <p><%= @user.bio %></p>
+</div>
+    <% @user.recent_posts.each do |post| %>
+        <div class="post-details">
+      <h2><%= link_to "Post #", user_post_url(@user.id, post.id) %><%= post.id %></h2>
+      <h3><%= post.text %></h3>
+      <p>Comments:<%= post.comments_counter %>,Likes:<%= post.likes_counter %></p>
+      </div>
+    <% end %>
+    <%= link_to user_posts_path(@user.id) do %>
+    <button type="button" class="see-posts">See all posts</button>
+    <% end %>
```

Comment:
- [ ] Nice work trying to get the recent posts that a user has with the method `recent_posts` :+1: But there is not a method named `recent_posts` in the User model and this will raise an `undefined method` error. Kindly create the `recent_posts` method to get the user's 3 most recent posts 
**HIT:** The method `recent_posts` could look like this :point_down: 
```ruby
# app/models/user.rb
def recent_posts
  posts.order(created_at: :desc).limit(3)
end
```